tests/migration_test.go

```diff
@@ -932,23 +933,45 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 		Context("with an Fedora shared NFS PVC, cloud init and service account", func() {
 			var pvName string
 			var vmi *v1.VirtualMachineInstance
-			BeforeEach(func() {
+
+			initNFS := func(isIpv6 bool) {
 				tests.SkipNFSTestIfRunnigOnKindInfra()
 				pvName = "test-nfs" + rand.String(48)
 				// Prepare a NFS backed PV
 				By("Starting an NFS POD")
 				os := string(cd.ContainerDiskFedora)
-				nfsIP := tests.CreateNFSTargetPOD(os)
+				nfsIPs := tests.CreateNFSTargetPOD(os)
 				// create a new PV and PVC (PVs can't be reused)
 				By("create a new NFS PV and PVC")
+				nfsIP := libnet.GetIp(libnet.GetPodIpsStrings(nfsIPs), isIpv6)
+				if nfsIP == "" && isIpv6 {
+					Skip("skipping ipv6 nfs test on a single stack cluster")
+				}
+				Expect(nfsIP).NotTo(BeEmpty())
 				tests.CreateNFSPvAndPvc(pvName, "5Gi", nfsIP, os)
-			})
+			}
 
 			AfterEach(func() {
+				By("Deleting the VMI")
+				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
+
+				By("Waiting for VMI to disappear")
+				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
+
+				By("Deleting NFS pod")
```

Comment:
This is just cosmetics, but you could create a function for this.

Reply:
> I would change `table.DescribeTable` and `initNFS` to get a string to distinct IPv4 and IPv6 instead of a boolean

Can you please explain why?

Reply:
I can share why it may help someone like me: Because as a reader, I may not be interested to see the details of how all this is done, just that it is logically done. Allowing me to examine the test from higher ground before getting dirty in more details.

On another matter, please note that the cleanup here seems unsafe. If anything fails to be cleaned, the others won't get a chance to get cleaned and the test may leave leftovers.

Reply:
done. Each removal was moved to a separate `AfterEach`.

Reply:
i think a one safe AfterEach would be better, one that allows each resource to be cleaned, without dependency of the other